src/Features/CSharp/Portable/ValidateFormatString/ValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,119 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.ValidateFormatString;
+
+namespace Microsoft.CodeAnalysis.CSharp.ValidateFormatString
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class ValidateFormatStringDiagnosticAnalyzer : AbstractValidateFormatStringDiagnosticAnalyzer
+    {
+        public override void Initialize(AnalysisContext context)
+        {
+            context.RegisterSyntaxNodeAction(AnalyzeNode, SyntaxKind.InvocationExpression);
+        }
+
+        private static void AnalyzeNode(SyntaxNodeAnalysisContext context)
+        {
+            var invocationExpr = (InvocationExpressionSyntax)context.Node;
+            if (invocationExpr.ArgumentList?.Arguments == null)
+            {
+                return;
+            }
+
+            SimpleNameSyntax name;
+            // When calling string.Format(...), Format will be of type MemberAccessExpressionSyntax 
+            if (invocationExpr.Expression is MemberAccessExpressionSyntax memberAccessExpr)
+            {
+                if (memberAccessExpr.Name?.ToString() != "Format")
```

Comment:
:memo: For those unaware, the importance of `ValueText` comes into play especially for special cases of identifier handling in C#. For example, the following are not only all valid, but they are all the same in C#:

```
string.Format("{0}", 1);
string.@Format("{0}", 1);
string.F\u006frmat("{0}", 1);
```

While the `Text` property (and `ToString`) will return the text *as written* (3 different ways here), the `ValueText` property will return the interpreted text, which in all three cases is `Format`.